packages/core/src/stylable-transformer.ts

```diff
@@ -578,7 +586,10 @@ export class StylableTransformer {
                             : node.nodes
                     ) as Selector[];
                     const nestedContext = context.createNestedContext(innerSelectors);
-                    this.scopeSelectorAst(nestedContext);
+                    const transformedSelectors = this.scopeSelectorAst(nestedContext);
+                    for (let i = 0; i < transformedSelectors.length; i++) {
```

Comment:
It bugs me that `scopeSelectorAst` is mutative to the inner selector nodes, but not to the selector itself. I would prefer it would mutate the entire selector. In the current state this patch will probably be required in other places as well.

Lets see if we can change the input selector that goes into `scopeSelectorAst`